DuckDuckGo/TabViewControllerBrowsingMenuExtension.swift

```diff
@@ -236,37 +251,30 @@ extension TabViewController {
                                               image: UIImage(named: "MenuFavorite")!,
                                               action: { [weak self] in
             Pixel.fire(pixel: addToFavoriteFlow ? .browsingMenuAddToFavoritesAddFavoriteFlow : .browsingMenuAddToFavorites)
-            self?.performSaveFavoriteAction(for: link)
+            self?.performAddFavoriteAction(for: link, with: bookmarksInterface)
         })
         return entry
     }
     
-    private func performSaveFavoriteAction(for link: Link) {
-        bookmarksManager.saveNewFavorite(withTitle: link.title ?? "", url: link.url) { _, _ in
-            DispatchQueue.main.async {
-                ActionMessageView.present(message: UserText.webSaveFavoriteDone, actionTitle: UserText.actionGenericUndo, onAction: {
-                    self.performRemoveFavoriteAction(for: link)
-                })
-            }
-        }
+    private func performAddFavoriteAction(for link: Link,
+                                          with bookmarksInterface: MenuBookmarksInteracting) {
+        bookmarksInterface.createOrToggleFavorite(title: link.title ?? "", url: link.url)
```

Comment:
It's possible for this to be out of sync when you're loading a new page - I was able to get one bookmark with a blank title, and another with the title of the previous page. However, this isn't a regression introduced in this PR, and it takes fairly precise timing, I don't consider it a blocker.

Reply:
Will create a separate task for this, thanks.

Reply:
https://app.asana.com/0/414709148257752/1203462254653688/f